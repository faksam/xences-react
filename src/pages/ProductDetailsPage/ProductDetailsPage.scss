.ActiveProductImg {
  height: 18rem;
}

.ActiveProductImgCol {
  height: 5rem;
}
