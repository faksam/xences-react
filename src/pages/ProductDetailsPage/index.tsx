// import libraries
import * as React from 'react';

// third-party libraries
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

// styles
import './ProductDetailsPage.scss';

// components
import {
  Button,
  Card,
  Checkbox,
  Col,
  Collapsible,
  CollapsibleItem,
  Collection,
  CollectionItem,
  Row,
  Select,
  TextInput,
} from 'react-materialize';
import ProductCard from '../../components/ProductCard';

// interfaces

// helper functions

// fixtures
import { brands, colors } from '../../store/modules/brands/fixtures';
import { categories } from '../../store/modules/categories/fixtures';
import { menClothings, products } from '../../store/modules/products/fixtures';

export class ProductDetailsPage extends React.Component<{}> {
  render() {
    const product = products[1];
    return (
      <React.Fragment>
        <Row>
          <Col s={2}>
            <Row>
              <img className="right ActiveProductImgCol" src={product.pictures} />
            </Row>
            <Row>
              <img className="right ActiveProductImgCol" src={product.pictures} />
            </Row>
            <Row>
              <img className="right ActiveProductImgCol" src={product.pictures} />
            </Row>
            <Row>
              <img className="right ActiveProductImgCol" src={product.pictures} />
            </Row>
          </Col>
          <Col s={4}>
            <img className="ActiveProductImg" src={product.pictures} />
          </Col>
          <Col s={2}>
            <h5><b>{product.name}</b></h5>
            <p>{product.category}</p>
            <p>&#8358;{product.price}</p>
          </Col>
          <Col s={4} className="center">
            <Card>
              <Row>
                <Select s={3} label="Size">
                  <option value="S">S</option>
                  <option value="M">M</option>
                  <option value="L">L</option>
                  <option value="XL">XL</option>
                  <option value="XXL">XXL</option>
                </Select>
                <TextInput label="Quantity" placeholder="1" type="number" max="100" min="1" />
                <Select s={5} label="Color">
                  <option value="White">White</option>
                  <option value="Black">Black</option>
                  <option value="Blue">Blue</option>
                  <option value="Green">Green</option>
                  <option value="Red">Red</option>
                </Select>
              </Row>
              <Row>
                <Button>ADD TO CART</Button>
              </Row>
            </Card>
          </Col>
        </Row>
      </React.Fragment>
    );
  }
}

export default ProductDetailsPage;
