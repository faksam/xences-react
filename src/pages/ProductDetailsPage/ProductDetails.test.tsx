// react libraries
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

// third-party libraries
import { mount } from 'enzyme';

// components
import ProductDetailsPage from './';

describe('The ProductDetailsPage component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Router>
        <ProductDetailsPage />
      </Router>
    );
  });

  it('should be render products page', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
