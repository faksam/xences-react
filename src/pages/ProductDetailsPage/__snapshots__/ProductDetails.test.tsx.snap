// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ProductDetailsPage component should be render products page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ProductDetailsPage>
      <Row>
        <div
          className="row"
        >
          <Col
            s={2}
          >
            <div
              className="col s2"
            >
              <Row>
                <div
                  className="row"
                >
                  <img
                    className="right ActiveProductImgCol"
                    src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                  />
                </div>
              </Row>
              <Row>
                <div
                  className="row"
                >
                  <img
                    className="right ActiveProductImgCol"
                    src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                  />
                </div>
              </Row>
              <Row>
                <div
                  className="row"
                >
                  <img
                    className="right ActiveProductImgCol"
                    src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                  />
                </div>
              </Row>
              <Row>
                <div
                  className="row"
                >
                  <img
                    className="right ActiveProductImgCol"
                    src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                  />
                </div>
              </Row>
            </div>
          </Col>
          <Col
            s={4}
          >
            <div
              className="col s4"
            >
              <img
                className="ActiveProductImg"
                src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
              />
            </div>
          </Col>
          <Col
            s={2}
          >
            <div
              className="col s2"
            >
              <h5>
                <b>
                  Nice cap
                </b>
              </h5>
              <p>
                Clothings
              </p>
              <p>
                â‚¦
                2000
              </p>
            </div>
          </Col>
          <Col
            className="center"
            s={4}
          >
            <div
              className="col s4 center"
            >
              <Card>
                <div
                  className="card"
                >
                  <div
                    className="card-content"
                  >
                    <div>
                      <Row>
                        <div
                          className="row"
                        >
                          <TextInput
                            label="Quantity"
                            max="100"
                            min="1"
                            placeholder="1"
                            type="number"
                          >
                            <div
                              className="input-field col"
                            >
                              <input
                                className=""
                                id={2}
                                max="100"
                                min="1"
                                placeholder="1"
                                type="number"
                              />
                              <label
                                className="active"
                                htmlFor={2}
                              >
                                Quantity
                              </label>
                              <span
                                className="helper-text"
                              />
                            </div>
                          </TextInput>
                          <Select
                            label="Size"
                            options={
                              Object {
                                "classes": "",
                                "dropdownOptions": Object {
                                  "alignment": "left",
                                  "autoTrigger": true,
                                  "closeOnClick": true,
                                  "constrainWidth": true,
                                  "container": null,
                                  "coverTrigger": true,
                                  "hover": false,
                                  "inDuration": 150,
                                  "onCloseEnd": null,
                                  "onCloseStart": null,
                                  "onOpenEnd": null,
                                  "onOpenStart": null,
                                  "outDuration": 250,
                                },
                              }
                            }
                            s={3}
                            value="Size"
                          >
                            <div
                              className="input-field col s3"
                            >
                              <select
                                className=""
                                id={3}
                                onChange={[Function]}
                                options={
                                  Object {
                                    "classes": "",
                                    "dropdownOptions": Object {
                                      "alignment": "left",
                                      "autoTrigger": true,
                                      "closeOnClick": true,
                                      "constrainWidth": true,
                                      "container": null,
                                      "coverTrigger": true,
                                      "hover": false,
                                      "inDuration": 150,
                                      "onCloseEnd": null,
                                      "onCloseStart": null,
                                      "onOpenEnd": null,
                                      "onOpenStart": null,
                                      "outDuration": 250,
                                    },
                                  }
                                }
                                type="select"
                                value="Size"
                              >
                                <option
                                  disabled={true}
                                  key="Size/.0"
                                  value="Size"
                                />
                                <option
                                  key="S/.1"
                                  value="S"
                                >
                                  S
                                </option>
                                <option
                                  key="M/.2"
                                  value="M"
                                >
                                  M
                                </option>
                                <option
                                  key="L/.3"
                                  value="L"
                                >
                                  L
                                </option>
                                <option
                                  key="XL/.4"
                                  value="XL"
                                >
                                  XL
                                </option>
                                <option
                                  key="XXL/.5"
                                  value="XXL"
                                >
                                  XXL
                                </option>
                              </select>
                              <label
                                htmlFor={3}
                              >
                                Size
                              </label>
                            </div>
                          </Select>
                          <Select
                            label="Color"
                            options={
                              Object {
                                "classes": "",
                                "dropdownOptions": Object {
                                  "alignment": "left",
                                  "autoTrigger": true,
                                  "closeOnClick": true,
                                  "constrainWidth": true,
                                  "container": null,
                                  "coverTrigger": true,
                                  "hover": false,
                                  "inDuration": 150,
                                  "onCloseEnd": null,
                                  "onCloseStart": null,
                                  "onOpenEnd": null,
                                  "onOpenStart": null,
                                  "outDuration": 250,
                                },
                              }
                            }
                            s={5}
                            value="Color"
                          >
                            <div
                              className="input-field col s5"
                            >
                              <select
                                className=""
                                id={4}
                                onChange={[Function]}
                                options={
                                  Object {
                                    "classes": "",
                                    "dropdownOptions": Object {
                                      "alignment": "left",
                                      "autoTrigger": true,
                                      "closeOnClick": true,
                                      "constrainWidth": true,
                                      "container": null,
                                      "coverTrigger": true,
                                      "hover": false,
                                      "inDuration": 150,
                                      "onCloseEnd": null,
                                      "onCloseStart": null,
                                      "onOpenEnd": null,
                                      "onOpenStart": null,
                                      "outDuration": 250,
                                    },
                                  }
                                }
                                type="select"
                                value="Color"
                              >
                                <option
                                  disabled={true}
                                  key="Color/.0"
                                  value="Color"
                                />
                                <option
                                  key="White/.1"
                                  value="White"
                                >
                                  White
                                </option>
                                <option
                                  key="Black/.2"
                                  value="Black"
                                >
                                  Black
                                </option>
                                <option
                                  key="Blue/.3"
                                  value="Blue"
                                >
                                  Blue
                                </option>
                                <option
                                  key="Green/.4"
                                  value="Green"
                                >
                                  Green
                                </option>
                                <option
                                  key="Red/.5"
                                  value="Red"
                                >
                                  Red
                                </option>
                              </select>
                              <label
                                htmlFor={4}
                              >
                                Color
                              </label>
                            </div>
                          </Select>
                        </div>
                      </Row>
                      <Row>
                        <div
                          className="row"
                        >
                          <Button
                            node="button"
                          >
                            <button
                              className="btn"
                              disabled={false}
                            >
                              ADD TO CART
                            </button>
                          </Button>
                        </div>
                      </Row>
                    </div>
                  </div>
                  <div
                    className="card-reveal"
                  >
                    <span
                      className="card-title"
                    >
                      <Icon
                        right={true}
                      >
                        <i
                          className="material-icons right"
                        >
                          close
                        </i>
                      </Icon>
                    </span>
                  </div>
                </div>
              </Card>
            </div>
          </Col>
        </div>
      </Row>
    </ProductDetailsPage>
  </Router>
</BrowserRouter>
`;
